core/time/timezone/tz_unix.odin

```diff
@@ -0,0 +1,91 @@
+#+build darwin, linux, freebsd, openbsd, netbsd
+package timezone
+
+import "core:os"
+import "core:strings"
+import "core:path/filepath"
+import "core:time/datetime"
+
+local_tz_name :: proc(allocator := context.allocator) -> (name: string, success: bool) {
```

Comment:
A lot of allocations in this proc don't use the given allocator (`os.absolute_path_from_relative`, `os.read_entire_file` etc.), I suggest doing `context.allocator = allocator` at the beginning so everything is definitely covered.

Additionally there are a bunch of allocations that are temporary in nature, it may make sense to take a `temp_allocator := context.temp_allocator` and do those allocations on that allocator instead, and just have the resulting name be allocated on the `allocator`.

You also don't handle any allocation errors here, which may be fine, but you may get `"", true` back from the function if the string clone failed for example.

Reply:
I've taken a big allocator fail pass on tzif, if you could give it a once-over that'd be super helpful.

I *think* what I'm doing is reasonable to cleanup, but let me know